feat(ui): redesign error states for 404/400 and fix status codes

saishmungase opened the pull request:
<!--
Thank you for contributing to Open Food Facts Explorer!
Please provide a description of your changes below.
-->

## Description

### Summary
This PR overhauls the error handling logic for the product page. Currently, standard API responses like "Product Not Found" or "Invalid Barcode" are treated as critical system failures (HTTP 500), showing an aggressive error screen to the user.

This update correctly maps these API states to HTTP 404 and 400 status codes and introduces friendly "Empty State" UIs to guide users to contribute or correct their search.

### Detailed Changes

#### 1. Backend Logic (`src/routes/product/[barcode]/+page.ts`)
- **Strict Type Definitions:** I added strict TypeScript interfaces (`OffMessage`, `ProductStateResponse`) to strictly parse the API error object. These types were adopted directly from the official [Open Food Facts API v3 Documentation](https://openfoodfacts.github.io/openfoodfacts-server/api/ref-v3/#get-/api/v3/product/-code-) to ensure accuracy.
- **Error Mapping:**
    - **404 Not Found:** When the API returns `result.id: "product_not_found"`, we now throw a 404 error instead of a generic 500.
    - **400 Bad Request:** When the API returns `message.id: "invalid_code"`, we now throw a 400 error.
- **Data Serialization:** Implemented a `cleanErrors` utility to remove undefined values from the error object before passing it to the frontend, preventing SvelteKit serialization issues.

#### 2. Component UI (`src/lib/ui/StandardError.svelte`)
I redesigned the generic error component to handle specific status codes with dedicated UI cards:
- **404 (Product):** Shows a **Kiwi (🥝)** card with the message "We don't know this product yet!" and a primary button to add the product via the Perl form.
- **400 (Barcode):** Shows a **Thinking Face (🤔)** card with the message "Let's check that barcode" to help users fix typos.
- **500 (Server Error):** Refactored the "System Crash" card. It now uses a solid red header (`bg-error`) with white text (`text-error-content`) to ensure readability in **Dark Mode**, and technical details (like `Impact ID`) are hidden inside a collapsible section to reduce user confusion.

#### 3. Global Error Layout (`src/routes/+error.svelte`)
- **Global 404:** Added a **Compass (🧭)** screen to handle invalid routes (e.g., `/random-url`). This logic (`isGlobal404`) distinguishes between a "Page Not Found" (wrong URL) and a "Product Not Found" (valid URL, missing data).
- **Layout Improvements:** Removed the forced `alert-error` wrapper from the main container. This allows the new `StandardError` component to render full-width, centered cards, while maintaining the specific alert styling only for network errors.
- **DX:** Suppressed `console.error` logging for standard 404 route errors to keep the developer console clean.

Fixes #870
Relates to #837

### Screenshots

| Scenario | New UI |
| :--- | :--- |
| **Product Not Found (404)** <br> *Prompts to add product* | ![Product Not Found](https://github.com/user-attachments/assets/dc350ea4-adbe-4117-a206-ec958dfbc0df) |
| **Invalid Barcode (400)** <br> *Helpful typo feedback* | ![Invalid Barcode](https://github.com/user-attachments/assets/862f8fe2-d5e3-4556-b2f8-d07c1e3df864) |
| **Global 404** <br> *Invalid URL* | ![Global 404](https://github.com/user-attachments/assets/8fbb8931-498d-43c2-875a-9df8291d3066) |
| **Server Error (500)** <br> *Actual crash* | ![Server Error](https://github.com/user-attachments/assets/accbe92e-5e74-427f-99a9-d5207843259d) |

---

## Checklist:

**Author Self-Review:**

- [x] I have performed a self-review of my own code.
- [x] I understand the changes I'm proposing and why they are needed.
- [x] My changes generate no new warnings or errors (linting, console).
- [x] I have made corresponding changes to the documentation (if applicable).

**LLM Usage Disclosure:**
_Please be transparent about the use of AI assistance._

- [x] If I **did** use an AI Large Language Model, I have reviewed the generated code/text to ensure its accuracy, security, and relevance to the project's context and licensing.

**Triggering Code Review:**

- You can request an AI-powered code review by commenting `/gemini review` on this PR after it's been created.

VaiTon commented on src/lib/ui/StandardError.svelte:
```diff
@@ -1,64 +1,128 @@
+	const status = page.status;
+	let { message, errors } = $derived(error);
+	let isProductError = $derived(
+		message === 'Product Not Found' || message === 'Invalid Barcode Format'
```
I don't really like having these strings hard-coded here, because if we modify the error string where it gets thrown this would cease to work.

VaiTon commented on src/lib/ui/StandardError.svelte:
```diff
@@ -1,64 +1,128 @@
+		<div class="card bg-base-100 border-error w-full max-w-2xl overflow-hidden border-2 shadow-xl">
+			<div class="bg-error flex items-center gap-4 p-6">
+				<div class="bg-base-100 text-error rounded-full p-3 shadow-sm">
+					<svg
```
Don't use embedded SVGs if possible. Prefer using https://iconify.design/, like we do in the rest of the codebase

saishmungase commented:
@VaiTon I Pushed a quick fix for the CI build errors.

The new strict Union Types introduced in the previous commit correctly exposed some unsafe property access that TypeScript wasn't catching before. I've resolved them:

1.  **`src/routes/products/[barcode]/+page.svelte`**: TypeScript was (correctly) flagging that `productState.product` doesn't exist if the status is `failure`. I added a strict status check to ensure we only access `.product` when the status is `success`.
2.  **`src/lib/ui/images/PhotoManager.svelte`**: Fixed a Svelte 5 reactivity warning (`state_referenced_locally`). I switched to using `untrack()` for initialization and an `$effect` to ensure the language state properly syncs when navigating between products.

Build should pass now!

VaiTon commented:
@saishmungase can you please fix lint errors?
